In [1]:
import pandas as pd
import numpy as np


In [2]:
df=pd.read_csv("AB_NYC_2019.csv")

In [3]:
df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [5]:
df.dropna(inplace= True)
df.drop_duplicates(inplace =True)

In [6]:
df["id"]=df["id"].astype(str)
df["host_id	"]=df["host_id"].astype(str)
df["latitude"]=df["latitude"].astype(str)
df["longitude"]=df["longitude"].astype(str)


In [7]:
df.describe()

host_id         price  minimum_nights  number_of_reviews  \
count  3.882100e+04  38821.000000    38821.000000       38821.000000   
mean   6.424582e+07    142.332526        5.869220          29.290255   
std    7.589752e+07    196.994756       17.389026          48.182900   
min    2.438000e+03      0.000000        1.000000           1.000000   
25%    7.029525e+06     69.000000        1.000000           3.000000   
50%    2.837092e+07    101.000000        2.000000           9.000000   
75%    1.018905e+08    170.000000        4.000000          33.000000   
max    2.738417e+08  10000.000000     1250.000000         629.000000   

       reviews_per_month  calculated_host_listings_count  availability_365  
count       38821.000000                    38821.000000      38821.000000  
mean            1.373229                        5.166611        114.886299  
std             1.680328                       26.302954        129.529950  
min             0.010000                        1.000000          0.000000  
25%             0.190000                        1.000000          0.000000  
50%             0.720000                        1.000000         55.000000  
75%             2.020000                        2.000000        229.000000  
max            58.500000                      327.000000        365.000000

range of minimum nights for listings is 1 and 1250

## **Categorical Data**

In [8]:
df.nunique()

id                                38821
name                              38253
host_id                           30232
host_name                          9885
neighbourhood_group                   5
neighbourhood                       218
latitude                          17436
longitude                         13639
room_type                             3
price                               581
minimum_nights                       89
number_of_reviews                   393
last_review                        1764
reviews_per_month                   937
calculated_host_listings_count       47
availability_365                    366
host_id\t                         30232
dtype: int64

In [9]:
df.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365', 'host_id\t'],
      dtype='object')

In [10]:
df["room_type"].value_counts()

room_type
Entire home/apt    20321
Private room       17654
Shared room          846
Name: count, dtype: int64

In [11]:
df["room_type"].value_counts(normalize = True)

room_type
Entire home/apt    0.523454
Private room       0.454754
Shared room        0.021792
Name: proportion, dtype: float64

In [12]:
df["neighbourhood_group"].value_counts()

neighbourhood_group
Manhattan        16621
Brooklyn         16439
Queens            4572
Bronx              875
Staten Island      314
Name: count, dtype: int64

## **Numerical Data**


In [13]:
df["price"].value_counts(bins = 5)

(-10.001, 2000.0]    38786
(2000.0, 4000.0]        20
(4000.0, 6000.0]         8
(8000.0, 10000.0]        5
(6000.0, 8000.0]         2
Name: count, dtype: int64

In [14]:
bins = [-10,0, 50,100, 200,500,800,2000,4000,10000]
df["price"].value_counts(bins = bins)

(50.0, 100.0]        14212
(100.0, 200.0]       13544
(200.0, 500.0]        5267
(0.0, 50.0]           5166
(500.0, 800.0]         404
(800.0, 2000.0]        183
(2000.0, 4000.0]        20
(4000.0, 10000.0]       15
(-10.001, 0.0]          10
Name: count, dtype: int64

It is mainly helpful in small datasets.

## **Measures of central tendency**


In [15]:
df["price"].mean()

np.float64(142.33252621004095)

In [16]:
df["price"].median()

101.0

In [17]:
df["price"].std()

196.99475591833985

In [18]:
df["minimum_nights"].mean()

np.float64(5.869220267381056)

In [20]:
df["minimum_nights"].median()

2.0

## **Measure of Spread**


In [21]:
df["price"].skew()

np.float64(23.673594295123014)

In [23]:
df["price"].kurt()

np.float64(953.4807356344944)

How many listings have availability throughout the year (365 days)

In [24]:
df[df["availability_365"]==365].shape[0]

841

In [26]:
df.select_dtypes(include=['number']).corr()

host_id     price  minimum_nights  \
host_id                         1.000000  0.006263       -0.051673   
price                           0.006263  1.000000        0.025501   
minimum_nights                 -0.051673  0.025501        1.000000   
number_of_reviews              -0.141973 -0.035924       -0.069366   
reviews_per_month               0.296274 -0.030623       -0.121712   
calculated_host_listings_count  0.149417  0.052895        0.073474   
availability_365                0.155361  0.078276        0.101658   

                                number_of_reviews  reviews_per_month  \
host_id                                 -0.141973           0.296274   
price                                   -0.035924          -0.030623   
minimum_nights                          -0.069366          -0.121712   
number_of_reviews                        1.000000           0.549699   
reviews_per_month                        0.549699           1.000000   
calculated_host_listings_count          -0.059796          -0.009442   
availability_365                         0.193409           0.185896   

                                calculated_host_listings_count  \
host_id                                               0.149417   
price                                                 0.052895   
minimum_nights                                        0.073474   
number_of_reviews                                    -0.059796   
reviews_per_month                                    -0.009442   
calculated_host_listings_count                        1.000000   
availability_365                                      0.182981   

                                availability_365  
host_id                                 0.155361  
price                                   0.078276  
minimum_nights                          0.101658  
number_of_reviews                       0.193409  
reviews_per_month                       0.185896  
calculated_host_listings_count          0.182981  
availability_365                        1.000000